x-pack/plugins/apm/public/components/app/correlations/correlations_log.tsx

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiAccordion, EuiCode, EuiPanel } from '@elastic/eui';
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+
+interface Props {
+  log: string[];
+  displayLog?: boolean;
+}
+export function CorrelationsLog({ log }: Props) {
+  return (
+    <EuiAccordion
+      id="apmCorrelationsLogAccordion"
+      buttonContent={i18n.translate('xpack.apm.correlations.logButtonContent', {
+        defaultMessage: 'Log',
+      })}
+    >
+      <EuiPanel color="subdued">
+        {log.map((d, i) => {
+          const splitItem = d.split(': ');
```

Comment:
This looks odd. Instead of splitting, can we return logs in a structured format? It looks like we concatenate them here:

https://github.com/elastic/kibana/blob/d155a94fcec95cb6f99e879b099652623fc80451/x-pack/plugins/apm/server/lib/search_strategies/correlations/async_search_service_log.ts#L28

I have some other minor suggestions for that file:
```ts
import { currentTimeAsString } from './utils';

interface LogMessage {
  timestamp: string;
  message: string;
  error?: string;
}

export const asyncSearchServiceLogProvider = () => {
  const logMessages: LogMessage[] = [];

  function addLogMessage(message: string, error?: string) {
    logMessages.push({ timestamp: currentTimeAsString(), message, error });
  }

  function getLogMessages() {
    return logMessages;
  }

  return { addLogMessage, getLogMessages };
};

export type AsyncSearchServiceLog = ReturnType<
  typeof asyncSearchServiceLogProvider
>;
```

Reply:
While diving into the above log stuff I noticed this section:

https://github.com/elastic/kibana/blob/d155a94fcec95cb6f99e879b099652623fc80451/x-pack/plugins/apm/server/lib/search_strategies/correlations/search_strategy.ts#L55-L64

That can be simplified to:
```ts
      if (request.id) {
        const existingGetAsyncSearchServiceState = asyncSearchServiceMap.get(
          request.id
        );

        if (existingGetAsyncSearchServiceState === undefined) {
          throw new Error(
            `AsyncSearchService with ID '${request.id}' does not exist.`
          );
        }
```

Reply:
This code/logic didn't change since 7.14, it was just moved to a separate component here to consolidate duplicate code. Moving to a structured log would require touching both server+client code again. Since we're past FF I'd like to suggest to leave it as is for 7.15. We already have an item for 7.16 to revisit this (`Log Error handling ...` item) here #109220

Reply:
Doing this for 7.16 is fine. Do you want to track this by referencing this comment? 

I also recall adding a comment about replacing the generator function with promises (preferably) or observables since that's what we already use and we should keep the number of async concepts down when possible. Can we add that to the list as well?

Reply:
👍  I linked back to this conversation from the meta issue's item. In 7.15 we moved the generator function to it's own file and added tests so it should be more safe to refactor from now on. I'll add an item about it too.